zcash_client_sqlite/src/wallet/transact.rs

```diff
@@ -617,17 +617,22 @@ mod tests {
             .unwrap();
 
             // Fetch the transaction from the database
-            let raw_tx: Vec<_> = db_write
+            let (raw_tx, block_height): (Vec<_>, BlockHeight) = db_write
                 .wallet_db
                 .conn
                 .query_row(
-                    "SELECT raw FROM transactions
-                    WHERE id_tx = ?",
+                    "SELECT raw, block FROM transactions
+                WHERE id_tx = ?",
                     &[tx_row],
-                    |row| row.get(0),
+                    |row| {
+                        let h: u32 = row.get(1)?;
+                        Ok((row.get(0)?, BlockHeight::from(h)))
+                    },
                 )
                 .unwrap();
-            let tx = Transaction::read(&raw_tx[..]).unwrap();
+
+            let tx = Transaction::read(&raw_tx[..], BranchId::for_height(&network, block_height))
+                .unwrap();
```

Comment:
The changes in this file are also unnecessary once the bug I identified is fixed.